engine/src/main/java/io/camunda/zeebe/engine/processing/message/PendingProcessMessageSubscriptionChecker.java

```diff
@@ -27,8 +26,8 @@
   private final MutablePendingProcessMessageSubscriptionState pendingState;
   private final long subscriptionTimeoutInMillis;
 
-  private ActorControl actor;
-  private ScheduledTimer timer;
+  private ProcessingScheduleService scheduleService;
+  private boolean timerRunning = false;
```

Comment:
🔧 Changing the value of `timerRunning` doesn't stop the timer from running, but stops it from being rescheduled the next time. Perhaps a better name for `timerRunning` is `isTimerSchedulingAllowed`, `shouldRescheduleTimer` or `shouldContinueToRescheduleTimer` or something similar.